scripts/prepare_gnomAD.py

```diff
@@ -0,0 +1,231 @@
+"""Subset and annotate gnomAD hail table"""
+
+import gnomad.utils.vep
+import hail as hl
+from typing import Union
+
+
+def get_worst_consequence_with_non_coding(ht):
+    """
+    copied and adapted from gnomad_lof
+    """
+
+    def get_worst_csq(
+        csq_list: hl.expr.ArrayExpression, protein_coding: bool
+    ) -> hl.struct:
+        lof = hl.null(hl.tstr)
+        no_lof_flags = hl.null(hl.tbool)
+        # lof_filters = hl.null(hl.tstr)
+        # lof_flags = hl.null(hl.tstr)
+        if protein_coding:
+            all_lofs = csq_list.map(lambda x: x.lof)
+            lof = hl.literal(['HC', 'OS', 'LC']).find(all_lofs.contains)
+            csq_list = hl.if_else(
+                hl.is_defined(lof), csq_list.filter(lambda x: x.lof == lof), csq_list
+            )
+            no_lof_flags = hl.or_missing(
+                hl.is_defined(lof),
+                csq_list.any(lambda x: (x.lof == lof) & hl.is_missing(x.lof_flags)),
+            )
+            # lof_filters = hl.delimit(hl.set(csq_list.map(lambda x: x.lof_filter).filter(lambda x: hl.is_defined(x))), '|')
+            # lof_flags = hl.delimit(hl.set(csq_list.map(lambda x: x.lof_flags).filter(lambda x: hl.is_defined(x))), '|')
+        all_csq_terms = csq_list.flatmap(lambda x: x.consequence_terms)
+        worst_csq = hl.literal(gnomad.utils.vep.CSQ_ORDER).find(all_csq_terms.contains)
+        return hl.struct(
+            worst_csq=worst_csq,
+            protein_coding=protein_coding,
+            lof=lof,
+            no_lof_flags=no_lof_flags,
+            # lof_filters=lof_filters,
+            # lof_flags=lof_flags
+        )
+
+    protein_coding = ht.vep.transcript_consequences.filter(
+        lambda x: x.biotype == 'protein_coding'
+    )
+    return ht.annotate(
+        **hl.case(missing_false=True)
+        .when(hl.len(protein_coding) > 0, get_worst_csq(protein_coding, True))
+        .when(
+            hl.len(ht.vep.transcript_consequences) > 0,
+            get_worst_csq(ht.vep.transcript_consequences, False),
+        )
+        .when(
+            hl.len(ht.vep.regulatory_feature_consequences) > 0,
+            get_worst_csq(ht.vep.regulatory_feature_consequences, False),
+        )
+        .when(
+            hl.len(ht.vep.motif_feature_consequences) > 0,
+            get_worst_csq(ht.vep.motif_feature_consequences, False),
+        )
+        .default(get_worst_csq(ht.vep.intergenic_consequences, False))
+    )
+
+
+def prepare_ht(ht, trimer: bool = False):
```

Comment:
A docstring with the purpose of the function would be fantastic!

Reply:
I copied the function from `gnomad_lof` directly. Added file and line number of the original in the docstring